WordPress/Classes/ViewRelated/Post/Scheduling/CalendarMonthView.swift

```diff
@@ -0,0 +1,165 @@
+import Foundation
+import Gridicons
+
+class CalendarMonthView: UIView {
+
+    private struct Constants {
+        static let rowHeight: CGFloat = 44
+        static let rowInsets = UIEdgeInsets(top: 0, left: 16, bottom: 0, right: 16)
+        static let calendarHeight: CGFloat = 240
+        static let calendarWidth: CGFloat = 375
+    }
+
+    weak var headerTitle: UILabel?
+    weak var forwardButton: UIButton?
+    weak var previousButton: UIButton?
+
+    var updated: ((Date) -> Void)?
+
+    private static var dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "MMMM, YYYY"
+        return formatter
+    }()
+
+    var selectedDate: Date? {
+        didSet {
+            if let date = selectedDate {
+                calendarCollectionView.selectDates([date])
+                calendarCollectionView.scrollToDate(date, animateScroll: false)
+            }
+        }
+    }
+
+    private let calendarCollectionView = CalendarCollectionView()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setup() {
+        calendarCollectionView.calDataSource.didScroll = { [weak self] dateSegment in
+            if let visibleDate = dateSegment.monthDates.first?.date {
+                self?.headerTitle?.text = CalendarMonthView.dateFormatter.string(from: visibleDate)
+            }
+        }
+        calendarCollectionView.calDataSource.didSelect = { [weak self] dateSegment in
+            self?.updated?(dateSegment)
+        }
+
+        let weekdaysHeaderView = makeWeekdayHeaderView()
+        let calendarHeaderView = makeCalendarHeaderView()
+
+        let stackView = UIStackView(arrangedSubviews: [
+            calendarHeaderView,
+            weekdaysHeaderView,
+            calendarCollectionView
+        ])
+        stackView.axis = .vertical
+        stackView.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
Did this and also a bit more refactoring to move to two UIStackView subclasses. This could probably still be cleaned up but I thought it was a good start, at least.